src/util/immediate_task_runner.cpp

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <util/task_runner.h>
+
+#include <functional>
+
+namespace util {
+
+void ImmediateTaskRunner::insert(std::function<void()> func)
+{
+    func();
```

Comment:
In commit "kernel: Remove dependency on CScheduler" (492ff2c504d1da3f4ae8704a5485a286ee76f0d9)

I think it would be clearer just to inline these methods and eliminate this file. If not eliminating this file, I would also consider it renaming it to just `util/task_runner.cpp` so it easier to find next to the header file.